src/fixate/_switching.py

```diff
@@ -87,15 +103,23 @@ def __or__(self, other: PinUpdate) -> PinUpdate:
 
 PinUpdateCallback = Callable[[PinUpdate, bool], None]
 
+S = TypeVar("S", bound=str)
 
-class VirtualMux:
+from types import get_original_bases, resolve_bases
+
+
+class VirtualMux(Generic[S]):
+    map_tree: Optional[TreeDef] = None
+    map_list: Optional[Sequence[Sequence[str]]] = None
     pin_list: PinList = ()
     clearing_time: float = 0.0
 
     ###########################################################################
     # These methods are the public API for the class
-
     def __init__(self, update_pins: Optional[PinUpdateCallback] = None):
+        # digest all the typing information if there is any to sete pin_list and map_list
+        self._digest_type_hints()
```

Comment:
I think Clint mentioned this before, and I think you said it should be back where it was, but the comment still doesn't look like it's in the right place to me. Shouldn't it be above `self._pin_set`?